src/HealthChecks.MySql/MySqlHealthCheckOptions.cs

```diff
@@ -10,14 +9,19 @@ namespace HealthChecks.MySql;
 public class MySqlHealthCheckOptions
 {
     /// <summary>
-    /// The MySQL connection string to be used.
+    /// The MySQL data source to be used. This is the preferred way to specify the MySQL server to be checked.
     /// </summary>
-    public string ConnectionString { get; set; } = null!;
+    public MySqlDataSource? DataSource { get; set; }
```

Comment:
Exposing a `DataSource` can lead to issues similar to the ones I've fixed in #2045 for PostgreSQL. IMO we should make it internal.

```suggestion
    internal MySqlDataSource? DataSource { get; set; }
```

Reply:
> Let's make the parameterless MySqlHealthCheckOptions ctor internal ... Moreover, let's make the setters for these properties internal.

When developing this, I found that I didn't need a parameterless constructor nor to set the properties after construction. Thus, the `MySqlHealthCheckOptions` type has been updated to have just two public constructors and private setters for all properties.

Reply:
> When developing this, I found that I didn't need a parameterless constructor nor to set the properties after construction. Thus, the MySqlHealthCheckOptions type has been updated to have just two public constructors and private setters for all properties.

Great! 👍